src/main/java/com/github/kagkarlsson/scheduler/JdbcTaskRepository.java

```diff
@@ -92,7 +92,17 @@ public boolean createIfNotExists(Execution execution) {
 
 	@Override
 	public List<Execution> getDue(Instant now) {
-		return getDue(now, MAX_DUE_RESULTS);
+		return getDue(now, MAX_RESULTS);
+	}
+
+	@Override
+	public List<Execution> getScheduledExecutions() {
```

Comment:
My thought was that these methods should also take a consumer that is called for every row in the resultset, so  `ExecutionResultSetMapper` needs to be refactored (if possible) to take a consumer instead. (`getDue` should return a `List` though, but might use the refactored `ExecutionResultSetMapper` by sending in a consumer `mylist::add`)

Reply:
Oops, I must have forgotten about it. I updated the PR with also having the `JdbcTaskRepository` make use of the consumer. A wrapper class makes sure that the `getDue()` can still easily pass a mapper object in order to get back a list.